core/GlobalState.cc

```diff
@@ -580,6 +580,21 @@ void GlobalState::initEmpty() {
                  .untypedArg(Names::arg1())
                  .buildWithResultUntyped();
 
+    // Synthesize <Magic>.<to-hash-dup>(arg: T.untyped) => T.untyped
+    method = enterMethod(*this, Symbols::MagicSingleton(), Names::toHashDup())
+                 .untypedArg(Names::arg0())
+                 .buildWithResultUntyped();
+
+    // Synthesize <Magic>.<to-hash-nodup>(arg: T.untyped) => T.untyped
+    method = enterMethod(*this, Symbols::MagicSingleton(), Names::toHashNoDup())
+                 .untypedArg(Names::arg0())
+                 .buildWithResultUntyped();
```

Comment:
I'm having second thoughts about this being necessary: we could just fall back on `x.to_hash` instead.